.github/workflows/build-docs.yml

```diff
@@ -9,23 +9,46 @@ jobs:
   build:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
-    - name: Set up Python 3.9
-      uses: actions/setup-python@v2
-      with:
-        python-version: 3.9
-    - name: Install dependencies
-      run: |
-        python -m pip install --upgrade pip
-        pip install -r requirements.txt
-        pip install -r requirements_dev.txt
-        pip install -e .
-    - name: Build documentation
-      run: |
-        python -m sphinx -T -b html -d docs/_build/doctrees -D language=en docs docs/_build/html -n -W
-    - uses: actions/upload-artifact@v2
-      name: Upload docs as artifact
-      with:
-        name: docs-html
-        path: './docs/_build/html'
-        if-no-files-found: error
+      - uses: actions/checkout@v2
+
+      - name: Set up Python 3.9
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.9
+
+      - name: Get full Python version
+        id: full-python-version
+        run: echo ::set-output name=version::$(python -c "import sys; print('-'.join(str(v) for v in sys.version_info))")
```

Comment:
why we need this step?

Reply:
In order to create a reliable hash key for the `.venv` directory using the full Python version:

```yaml
- name: Set up cache
   uses: actions/cache@v2
   id: cache
   with:
     path: .venv
     key: venv-${{ steps.full-python-version.outputs.version }}-${{ hashFiles('**/poetry.lock') }}
```

I've used [Poetry's own GitHub Actions workflows](https://github.com/python-poetry/poetry/blob/master/.github/workflows/main.yml) as a template, they're doing it like this.